aii-core/src/main/perl/Shellfe.pm

```diff
@@ -701,13 +690,39 @@ sub fetch_profiles
     return %h;
 }
 
-
+# Initiate the Parallel:ForkManager with requested threads if option is given
+sub init_pm 
+{
+    my ($self, $cmd, $responses) = @_;
+    if ($self->option('parallel')) {
+        my $pm = Parallel::ForkManager->new($self->option('parallel'));
+        $pm->run_on_finish ( # called before the first call to start()
+            sub {
+                my ($pid, $exit_code, $id, $esignal, $cdump, $data_struct_ref) = @_;
+                if ($exit_code) {
+                    $self->error("Error running $cmd on $id, exitcode $exit_code");
+                };
+                # retrieve data structure from child 
+                if (defined($data_struct_ref)) {
+                    $responses->{$id} = $data_struct_ref;
+                    $self->debug(5, "Running $cmd on $id had output"); 
+                }
+            }
+        );
+        return $pm;
+    } else {
+        return;
+    }
+}
+    
 # Wrapper to execute the commands in sorted manner
 no strict 'refs';
```

Comment:
euuuhhhhhh, what? this is way too much code to disable strict. best to move the `sub {` to its own function, and only disable strict check inside the lofloop and only for the asignment